pkg/controllers/securitypolicy/securitypolicy_controller.go

```diff
@@ -46,6 +47,7 @@ var (
 	ResultRequeueAfter5mins = common.ResultRequeueAfter5mins
 	MetricResType           = common.MetricResTypeSecurityPolicy
 	once                    sync.Once
+	EnableVPCNetwork        = false
```

Comment:
Make it a member variable of `SecurityPolicyReconciler`, declaring a global variable for something that can be updated at runtime could cause troubles to maintenance and tests and is bad for encapsulation.

Reply:
It is used by https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/securitypolicy/securitypolicy_controller.go#L380-L380, a bit not convenient to put it as a member variable.

Reply:
The link doesn't point to its usage. I found it myself but the reason doesn't really justify it, you can't make everything global variables to worsen the encapsulation to adapt code that may not have been encapsulated well.

In this specific case, I see at least 3 ways to make `reconcileSecurityPolicy` the member method of `SecurityPolicyReconciler`, by which it could access both `EnableVPCNetwork` and `Client` directly.
1. Leverage `handler.Funcs` when calling `Watches`
2. Leverage `handler.EnqueueRequestsFromMapFunc` when calling `Watches`
3. Pass the function `reconcileSecurityPolicy` or an interface that implements `reconcileSecurityPolicy` to `EnqueueRequestForNamespace`

1 and 2 make more sense.

Reply:
OK, got it! Thanks